service/service.go

```diff
@@ -42,52 +67,103 @@ func NewService(cfg *Config) (*Service, error) {
 			case <-s.roundsTicker():
 			}
 
-			if len(s.activeRound.commitments) == 0 && !s.cfg.ExecuteEmpty {
+			if len(s.openRound.commitments) == 0 && !s.cfg.ExecuteEmpty {
 				continue
 			}
 
-			s.prevRound = s.activeRound
+			s.prevRound = s.openRound
 
 			roundId++
-			s.activeRound = s.newRound(roundId)
+			s.openRound = s.newRound(roundId)
 			log.Infof("round %v opened", roundId)
 
 			// Close previous round and execute it.
 			go func() {
+				// TODO(moshababo): apply safe concurrency
 				r := s.prevRound
+				s.executingRounds[r.id] = r
+
 				err := r.close()
 				if err != nil {
 					log.Error(err)
 				}
 				log.Infof("round %v closed, executing...", r.id)
-				phi, err := r.execute()
+				err = r.execute()
 				if err != nil {
 					log.Error(err)
 				}
-				log.Infof("round %v executed, phi=%v", r.id, phi)
+
+				delete(s.executingRounds, r.id)
+				s.executedRounds[r.id] = r
+				log.Infof("round %v executed, phi=%v", r.id, r.nip.Phi)
 			}()
 		}
 	}()
 
 	return s, nil
 }
 
-func (s *Service) SubmitCommitment(c []byte) (*submitResponse, error) {
-	r := s.activeRound
+func (s *Service) Info() *InfoResponse {
+	res := new(InfoResponse)
+	res.openRoundId = s.openRound.id
+
+	ids := make([]int, 0, len(s.executingRounds))
+	for id := range s.executingRounds {
```

Comment:
must be interlocked with NuewService

Reply:
Sorry, I didn't understand what you mean.

Reply:
`executingRounds` is accessed here and in another go routine, please use a mutex when reading from same memory concurrently.

Reply:
Yes, i'll left a TODO for this in the other goroutine. I'll use the same mutex which will be applied there.